translate_3to2.go

```diff
@@ -0,0 +1,330 @@
+	if ref.Source == nil {
+		return
+	}
+	ret = &old.ConfigReference{}
+	ret.Source = strV(ref.Source)
+	ret.Verification.Hash = ref.Verification.Hash
+	return
+}
+
+func translateCfgRefs3to2(refs []new.ConfigReference) (ret []old.ConfigReference) {
+	for _, ref := range refs {
+		ret = append(ret, *translateCfgRef3to2(ref))
+	}
+	return
+}
+
+func translateCAs3to2(refs []new.CaReference) (ret []old.CaReference) {
+	for _, ref := range refs {
+		ret = append(ret, old.CaReference{
+			Source: ref.Source,
+			Verification: old.Verification{
+				Hash: ref.Verification.Hash,
+			},
+		})
+	}
+	return
+}
+
+func translateUsers3to2(users []new.PasswdUser) (ret []old.PasswdUser) {
+	for _, u := range users {
+		ret = append(ret, old.PasswdUser{
+			Name:              u.Name,
+			PasswordHash:      u.PasswordHash,
+			SSHAuthorizedKeys: translateUserSSH3to2(u.SSHAuthorizedKeys),
+			UID:               u.UID,
+			Gecos:             strV(u.Gecos),
+			HomeDir:           strV(u.HomeDir),
+			NoCreateHome:      boolV(u.NoCreateHome),
+			PrimaryGroup:      strV(u.PrimaryGroup),
+			Groups:            translateUserGroups3to2(u.Groups),
+			NoUserGroup:       boolV(u.NoUserGroup),
+			NoLogInit:         boolV(u.NoLogInit),
+			Shell:             strV(u.Shell),
+			System:            boolV(u.System),
+		})
+	}
+	return
+}
+
+func translateUserSSH3to2(in []new.SSHAuthorizedKey) (ret []old.SSHAuthorizedKey) {
+	for _, k := range in {
+		ret = append(ret, old.SSHAuthorizedKey(k))
+	}
+	return
+}
+
+func translateUserGroups3to2(in []new.Group) (ret []old.Group) {
+	for _, g := range in {
+		ret = append(ret, old.Group(g))
+	}
+	return
+}
+
+func translateGroups3to2(groups []new.PasswdGroup) (ret []old.PasswdGroup) {
+	for _, g := range groups {
+		ret = append(ret, old.PasswdGroup{
+			Name:         g.Name,
+			Gid:          g.Gid,
+			PasswordHash: strV(g.PasswordHash),
+			System:       boolV(g.System),
+		})
+	}
+	return
+}
+
+func translateUnits3to2(units []new.Unit) (ret []old.Unit) {
+	for _, u := range units {
+		ret = append(ret, old.Unit{
+			Name:     u.Name,
+			Enabled:  u.Enabled,
+			Mask:     boolV(u.Mask),
+			Contents: strV(u.Contents),
+			Dropins:  translateDropins3to2(u.Dropins),
+		})
+	}
+	return
+}
+
+func translateDropins3to2(dropins []new.Dropin) (ret []old.SystemdDropin) {
+	for _, d := range dropins {
+		ret = append(ret, old.SystemdDropin{
+			Name:     d.Name,
+			Contents: strV(d.Contents),
+		})
+	}
+	return
+}
+
+func translateDisks3to2(disks []new.Disk) (ret []old.Disk) {
+	for _, d := range disks {
+		ret = append(ret, old.Disk{
+			Device:     d.Device,
+			WipeTable:  boolV(d.WipeTable),
+			Partitions: translatePartitions3to2(d.Partitions),
+		})
+	}
+	return
+}
+
+func translatePartitions3to2(parts []new.Partition) (ret []old.Partition) {
+	for _, p := range parts {
+		ret = append(ret, old.Partition{
+			Label:    strV(p.Label),
+			Number:   p.Number,
+			Size:     intV(p.SizeMiB),
```

Comment:
Size is in sectors while SizeMiB is in MiB. You'll either want to fail, assume a sector size, or translate to 2.3.0 which has SizeMiB/StartMiB fields.